ext/standard/url.c

```diff
@@ -190,8 +190,8 @@ PHPAPI php_url *php_url_parse_ex(char const *str, int length)
 			memcpy(port_buf, p, (pp - p));
 			port_buf[pp - p] = '\0';
 			port = strtol(port_buf, NULL, 10);
-			if (port > 0 && port <= 65535) {
-				ret->port = (unsigned short) port;
+			if (port > 0 && port <= 4294967295) {
```

Comment:
> No, I meant that `port <= 4294967295` will always be true, if long has only 32bits (CMIIW).

Yes, it has to be and you are absolutely right. No upper bound at all is definitely the better approach here and 0 inclusive as lower bound.

> `zend_long` is available as of PHP 7.0 only, but the PR targets 5.6.

Missed that it targets 5.6. I guess this requires another PR against 7 then where it uses `zend_long` and `#ifdef` here to determine if it is 32 or 64 bit.